Assets/MixedRealityToolkit/UX/Scripts/Progress/ProgressIndicator.cs

```diff
@@ -0,0 +1,315 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.XR.WSA.Input;
+using MixedRealityToolkit.Utilities.Solvers;
+using UnityEngine.XR.WSA;
+
+namespace MixedRealityToolkit.UX.Progress
+{
+    public abstract class Singleton<T> : MonoBehaviour where T : Singleton<T>
```

Comment:
We're adding another type of Singleton into the toolkit?

Not sure this is a good idea.